### PR TITLE
Update the benchmarks

### PR DESCRIPTION
**Pull request by martintmk**
### Details on the issue fix or feature implementation

Update the benchmarks. 

### Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [x]  I have included unit tests for the issue/feature
- [x]  I have successfully run a local build

### CODE REVIEW

File: BenchmarkDotNet.Artifacts/results/Polly.Core.Benchmarks.HedgingBenchmark-report-github.md [hunk 1]
```diff
@@ -9,9 +9,9 @@ Job=MediumRun  Toolchain=InProcessEmitToolchain  IterationCount=15
...
-|             Hedging_Primary |   941.7 ns |   5.45 ns |   7.81 ns |  1.00 |    0.00 |      - |      - |      40 B |        1.00 |
-|           Hedging_Secondary | 1,679.7 ns |   7.23 ns |  10.83 ns |  1.78 |    0.02 | 0.0076 |      - |     200 B |        5.00 |
-|   Hedging_Primary_AsyncWork | 4,867.5 ns |  71.40 ns | 102.40 ns |  5.17 |    0.12 | 0.0534 | 0.0229 |    1407 B |       35.17 |
-| Hedging_Secondary_AsyncWork | 7,093.2 ns | 234.32 ns | 343.47 ns |  7.51 |    0.37 | 0.0687 | 0.0610 |    1737 B |       43.42 |
+|                      Method |      Mean |     Error |    StdDev | Ratio | RatioSD |   Gen0 | Allocated | Alloc Ratio |
+|---------------------------- |----------:|----------:|----------:|------:|--------:|-------:|----------:|------------:|
+|             Hedging_Primary |  1.045 μs | 0.0049 μs | 0.0074 μs |  1.00 |    0.00 |      - |      40 B |        1.00 |
+|           Hedging_Secondary |  1.836 μs | 0.0043 μs | 0.0063 μs |  1.76 |    0.01 | 0.0076 |     224 B |        5.60 |
+|   Hedging_Primary_AsyncWork | 15.291 μs | 1.7958 μs | 2.6878 μs | 14.63 |    2.55 | 0.1678 |    4327 B |      108.17 |
```
Comment (martintmk) [inline BenchmarkDotNet.Artifacts/results/Polly.Core.Benchmarks.HedgingBenchmark-report-github.md:16]: This is a big drop,  however, I suspect this is caused by something on my VM as on main laptop the numbers are very similar to what it was before.

File: BenchmarkDotNet.Artifacts/results/Polly.Core.Benchmarks.CircuitBreakerOpenedBenchmark-report-github.md [hunk 2]
```diff
@@ -9,8 +9,8 @@ Job=MediumRun  Toolchain=InProcessEmitToolchain  IterationCount=15
...
 
 ```
-|                    Method |        Mean |     Error |    StdDev |      Median | Ratio | RatioSD |   Gen0 | Allocated | Alloc Ratio |
-|-------------------------- |------------:|----------:|----------:|------------:|------:|--------:|-------:|----------:|------------:|
-| ExecuteAsync_Exception_V7 | 29,720.5 ns | 177.63 ns | 265.86 ns | 29,629.0 ns | 56.63 |    0.58 | 0.0610 |    2888 B |       15.04 |
-| ExecuteAsync_Exception_V8 | 20,840.2 ns | 191.44 ns | 274.56 ns | 20,696.3 ns | 39.71 |    0.51 | 0.0610 |    1816 B |        9.46 |
-|   ExecuteAsync_Outcome_V8 |    525.1 ns |   1.07 ns |   1.47 ns |    525.0 ns |  1.00 |    0.00 | 0.0076 |     192 B |        1.00 |
+|                    Method |         Mean |     Error |    StdDev |  Ratio | RatioSD |   Gen0 | Allocated | Alloc Ratio |
+|-------------------------- |-------------:|----------:|----------:|-------:|--------:|-------:|----------:|------------:|
+| ExecuteAsync_Exception_V7 | 125,329.4 ns | 167.90 ns | 251.31 ns | 225.33 |    0.72 |      - |    2888 B |       15.04 |
```
Comment (martincostello) [inline BenchmarkDotNet.Artifacts/results/Polly.Core.Benchmarks.CircuitBreakerOpenedBenchmark-report-github.md:14]: These ones are pretty big changes too.

Could you spin up a VM in Azure or something so the baseline is a bit more stable and run there?

Comment (martintmk) [reply BenchmarkDotNet.Artifacts/results/Polly.Core.Benchmarks.CircuitBreakerOpenedBenchmark-report-github.md:14]: These are the numbers I get on my laptop:

image

Maybe I can just update those? We are mostly interested in allocations anyway. It's strange though, something must have happened on my devbox, I don't know the reason why it's reporting these inflated numbers.

Comment (martintmk) [reply BenchmarkDotNet.Artifacts/results/Polly.Core.Benchmarks.CircuitBreakerOpenedBenchmark-report-github.md]: Uploaded laptop-based ones.

**Review by martincostello — APPROVED** [review]

**Pull request merged**